cmd/xl-v1-object.go

```diff
@@ -774,21 +774,16 @@ func (xl xlObjects) putObject(bucket string, object string, data *hash.Reader, m
 func (xl xlObjects) deleteObject(bucket, object string) error {
 	// Initialize sync waitgroup.
 	var wg = &sync.WaitGroup{}
-
 	var writeQuorum int
 	var err error
-	// If its a directory request, no need to read metadata.
-	if !hasSuffix(object, slashSeparator) {
-		// Read metadata associated with the object from all disks.
-		metaArr, errs := readAllXLMetadata(xl.getDisks(), bucket, object)
```

Comment:
is this change still needed?

Reply:
> is this change still needed?

This is a non functional change, I just wanted to separate removing object & removing directory

Reply:
My suggestion is to avoid that in this PR @vadmeste

Reply:
> My suggestion is to avoid that in this PR @vadmeste

I also wanted to send a more minimal change but it turned out this is the simplest way to do that without being controversial.

Reply:
we already know we support directory operations cleanup is not needed IMHO simply fix the bug you can do cleanup later as well.